### ISSUE TITLE
Feature .id, .geometry, .properties, .type properties

### ISSUE THREAD

**Issue by sgillies** [geojson/py-feature-sequences#4]
Lately I'm feeling that, for example, `ftr.properties['foo']` is cleaner syntax than `ftr['properties']['foo']` and that we could surface all the standard GeoJSON items as Python properties in addition to the old item getter style. So, `ftr.id == ftr['id']` would be `True`.

We could get this by requiring the [geojson](https://github.com/frewsxcv/python-geojson) module or could roll our own dict-based classes.

**Comment by geowurster** [comment]
@sgillies Definitely cleaner syntax and more Pythonic but I wonder if it will cause confusion when some GeoJSON libraries have a `Feature()` class and others don't?

**Comment by sgillies** [comment]
We could update the Python geo protocol while we're at it. Would head off some confusion and help clear up the remainder.

**Comment by geowurster** [comment]
Yeah that and recommending people check for `__geo_interface__()` rather than assuming that an object that appears to be GeoJSON actually is, like our potential `Feature()` and `Geometry()` classes.

I sketched out a bit of this ticket in #1.
